### PR TITLE
Enhancement:  Support for zVM  compute nodes to Hosted Control Plane - vswitch FCP/DASD

### PR DESCRIPTION
**Pull request by veera-damisetti**
Enhancement:  Support for zVM  compute nodes to Hosted Control Plane - vswitch FCP/DASD

- Code changes for supporting zVM for Hosted Control Plane
- Supported network type  : vswitch
- Supported disk types: FCP/DASD
- Updated the documentation for the same

### CODE REVIEW

**Review by AmadeusPodvratnik — CHANGES_REQUESTED** [review]

File: inventories/default/group_vars/all.yaml.template [hunk 1]
```diff
@@ -239,13 +240,15 @@ rhcos_live_rootfs: "rhcos-4.12.3-s390x-live-rootfs.s390x.img"
 # Section 16 - Hypershift ( Optional )
 
 hypershift:
+  compute_node_type:       # zkvm / zvm
```
[the remark this answers is not in the document]
Comment (veera-damisetti) [reply inventories/default/group_vars/all.yaml.template]: Done

File: inventories/default/group_vars/all.yaml.template [hunk 2]
```diff
@@ -291,20 +297,76 @@ hypershift:
     mce_namespace: multicluster-engine # This is the Recommended Namespace for Multicluster Engine operator
 
   agents_parms:
+    agents_count:
+     
+    # zkvm specific parameters
```
[the remark this answers is not in the document]
Comment (veera-damisetti) [reply inventories/default/group_vars/all.yaml.template]: Done

File: roles/boot_zvm_nodes_hypershift/tasks/main.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        --zvmpass "{{ hypershift.agents_parms.zvm_parameters.nodes[item].password }}" \
+        --cpu "{{ hypershift.agents_parms.zvm_parameters.vcpus }}" \
+        --memory "{{ hypershift.agents_parms.zvm_parameters.memory }}" \
+        --kernel 'file:///var/lib/libvirt/images/pxeboot/kernel.img'  \
+        --initrd 'file:///var/lib/libvirt/images/pxeboot/initrd.img' \
+        --cmdline "$(cat /root/ansible_workdir/agent-{{ item }}.parm)"
+
+    - name:  Attaching dasd disk
+      shell: vmcp attach {{ hypershift.agents_parms.zvm_parameters.nodes[item].dasd.disk_id }} to {{ hypershift.agents_parms.zvm_parameters.nodes[item].name }} 
+      when: "{{ hypershift.agents_parms.zvm_parameters.disk_type == 'dasd' }}"
```
[the remark this answers is not in the document]
Comment (veera-damisetti) [reply roles/boot_zvm_nodes_hypershift/tasks/main.yaml]: Done

File: roles/setup_for_agents_hypershift/templates/parm-file.parm.j2 [hunk 4]
```diff
@@ -0,0 +1 @@
+rd.neednet=1 console=ttysclp0 coreos.live.rootfs_url=http://{{ hypershift.bastion_hypershift }}:8080/rootfs.img ip={{ hypershift.agents_parms.zvm_parameters.nodes[item].osa.ip }}::{{ hypershift.agents_parms.zvm_parameters.gateway }}:{{ hypershift.agents_parms.zvm_parameters.subnetmask }}::{{ hypershift.agents_parms.zvm_parameters.nodes[item].osa.ifname }}:none nameserver={{ hypershift.agents_parms.zvm_parameters.nameserver }} zfcp.allow_lun_scan=0 rd.znet=qeth,{{ hypershift.agents_parms.zvm_parameters.nodes[item].osa.id }},layer2=1 {% if hypershift.agents_parms.zvm_parameters.disk_type != 'fcp' %}rd.dasd=0.0.{{ hypershift.agents_parms.zvm_parameters.nodes[item].dasd.disk_id }}{% else %}rd.zfcp={{ hypershift.agents_parms.zvm_parameters.nodes[item].lun[0].paths[0].fcp}},{{ hypershift.agents_parms.zvm_parameters.nodes[item].lun[0].paths[0].wwpn }},{{ hypershift.agents_parms.zvm_parameters.nodes[item].lun[0].id }} {% endif %} random.trust_cpu=on rd.luks.options=discard ignition.firstboot ignition.platform.id=metal console=tty1 console=ttyS1,115200n8 coreos.inst.persistent-kargs="console=tty1 console=ttyS1,115200n8"
```
[the remark this answers is not in the document]
Comment (veera-damisetti) [reply roles/setup_for_agents_hypershift/templates/parm-file.parm.j2]: Done

File: playbooks/create_hosted_cluster.yaml [hunk 5]
```diff
@@ -8,9 +8,12 @@
     - name: Setting host
       set_fact:
         host: 'kvm_host_hypershift'
+      when: compute_node_type != 'zvm'
```
[the remark this answers is not in the document]
Comment (veera-damisetti) [reply playbooks/create_hosted_cluster.yaml]: Done

**Review by AmadeusPodvratnik — APPROVED** [review]
Looks good so far. I believe that for lpar support some changes are needed.

**Pull request merged**